### PR TITLE
[LETS-31] Part II load_trantable implementation

### PR DESCRIPTION
**Pull request by adrianbrab**

### CODE REVIEW

File: src/transaction/log_checkpoint_info.cpp [hunk 1]
```diff
@@ -16,6 +16,14 @@
...
 
+#include "log_impl.h"
+#include "client_credentials.hpp"
+#include "critical_section.h"
+#include "memory_alloc.h"
+#include "transaction_global.hpp"
+#include "thread_entry.hpp"
+#include "scope_exit.hpp"
+
 #include "log_checkpoint_info.hpp"
```
Comment (ribram) [inline src/transaction/log_checkpoint_info.cpp]: Includes format:

Include own header first.

Alphabetical order of project includes.

Alphabetical order of system includes.

File: src/transaction/log_checkpoint_info.cpp [hunk 2]
```diff
@@ -143,32 +151,135 @@ namespace cublog
...
+
+    //ENTER the critical section
+    TR_TABLE_CS_ENTER (thread_p);
+    log_Gl.prior_info.prior_lsa_mutex.lock ();
+
+    scope_exit <std::function<void (void)>> unlock_on_exit ([thread_p] ()
+    {
+      log_Gl.prior_info.prior_lsa_mutex.lock ();
+      TR_TABLE_CS_EXIT (thread_p);
+    });
```
Comment (ribram) [inline src/transaction/log_checkpoint_info.cpp:261]: Does it work like this?
```suggestion
    scope_exit unlock_on_exit ([thread_p] ()
    {
      log_Gl.prior_info.prior_lsa_mutex.lock ();
      TR_TABLE_CS_EXIT (thread_p);
    });
```

File: src/transaction/log_checkpoint_info.hpp [hunk 3]
```diff
@@ -17,8 +17,12 @@
  */
 
 #include "log_lsa.hpp"
-#include "log_record.hpp"
+#include "log_system_tran.hpp"
 #include "packable_object.hpp"
+#include "page_buffer.h"
+#include "system_parameter.h"
```
Comment (ribram) [inline src/transaction/log_checkpoint_info.hpp]: I don't think all includes are needed. log_system_tran.hpp, page_buffer.h and system_parameter.h shouldn't be required in this header.

File: src/transaction/log_record.hpp [hunk 4]
```diff
@@ -32,7 +32,7 @@
 #include "storage_common.h"
 #include "system.h"
 
-#include <string.h>
+#include <cstring>
```
Comment (ribram) [inline src/transaction/log_record.hpp]: Remove include.

File: unit_tests/log_transfer/test_main_chkpt_info.cpp [hunk 5]
```diff
@@ -527,5 +527,116 @@ namespace cubpacking
...
+prm_get_bool_value (PARAM_ID prmid)
+{
+  return false;
+}
+
+const char *
+clientids::get_db_user () const
+{
+  char *c;
+  return c;
```
Comment (ribram) [inline unit_tests/log_transfer/test_main_chkpt_info.cpp]:
```suggestion
  return nullptr;
```

File: src/transaction/log_checkpoint_info.hpp (hunk 3 above)
Comment (adrianbrab) [reply src/transaction/log_checkpoint_info.hpp]: log_system_trans is required for the function definitions of load_checkpoint_trans and load_checkpoint_topop, the rest have been moved

**Pull request merged**